### ISSUE TITLE
`experimental_strict_kotlin_deps` is complaining about the standard library

### ISSUE THREAD

**Issue by JeffreyFalgout** [bazelbuild/rules_kotlin#930]
```sh
$ bazel build <bazel target>
...
 ** Please add the following dependencies:
<snip>/external/com_github_jetbrains_kotlin/lib/kotlin-stdlib.jar to <bazel target>Target <bazel target> failed to build
```

 This error persists even if I add a dep on that jar via ` @com_github_jetbrains_kotlin//:kotlin-stdlib`

Is this intentional behavior?  I would've thought that the standard library would've been expected to be used by all targets by default

I'm not seeing anyone else complain about this, which makes me think that this is maybe something that we've done wrong in our WORKSPACE by accident. I'm not entirely sure which information will be relevant, but I'll try to seed some things that might help

```
rules_kotlin_version = 1.7.1
rules_jvm_external_version = version 4.5

maven_install(
  artifacts = [
      ...
     "org.jetbrains.kotlin:kotlin-stdlib-jdk7:jar:1.7.10",
     ...
  ],
  ...
)
```

```bzl
define_kt_toolchain(
    name = "kotlin_toolchain",
    experimental_strict_kotlin_deps = "error",
)

load("@io_bazel_rules_kotlin//kotlin:repositories.bzl", "kotlin_repositories")

kotlin_repositories()

register_toolchains("//:kotlin_toolchain")
```

**Comment by JeffreyFalgout** [comment]
I managed to rule out most of our WORKSPACE as the culprit:

```sh
$ cat WORKSPACE.bazel 
workspace(
    name = "testing",
)

load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

rules_kotlin_version = "1.7.1"

rules_kotlin_sha = "fd92a98bd8a8f0e1cdcb490b93f5acef1f1727ed992571232d33de42395ca9b3"

http_archive(
    name = "io_bazel_rules_kotlin",
    sha256 = rules_kotlin_sha,
    urls = ["https://github.com/bazelbuild/rules_kotlin/releases/download/v%s/rules_kotlin_release.tgz" % rules_kotlin_version],
)

load("@io_bazel_rules_kotlin//kotlin:repositories.bzl", "kotlin_repositories")

kotlin_repositories()

register_toolchains("//:kotlin_toolchain")

$ cat BUILD.bazel 
load("@io_bazel_rules_kotlin//kotlin:core.bzl", "define_kt_toolchain")
load("@io_bazel_rules_kotlin//kotlin:jvm.bzl", "kt_jvm_library")

define_kt_toolchain(
    name = "kotlin_toolchain",
    experimental_strict_kotlin_deps = "error",
)

kt_jvm_library(
    name = "foo",
    srcs = ["foo.kt"],
)

$ cat foo.kt 
package testing

fun emptyToNull(s: String): String? {
  if (s.isEmpty()) {
    return nil
  }
  return s
}

$ bazel build :foo
...
/usr/local/google/home/jfalgout/.cache/bazel/_bazel_jfalgout/abaf86743bc1a60bd72a4b4cb42f49bb/execroot/testing/external/com_github_jetbrains_kotlin/lib/kotlin-stdlib.jar to //:fooTarget //:foo failed to build
```

**Comment by JeffreyFalgout** [comment]
For your convenience: https://github.com/JeffreyFalgout/playground/tree/main/rules_kotlin/930
